lib/widgets/feedback_question.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:acs_upb_mobile/pages/class_feedback/model/questions/question.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/model/questions/question_dropdown.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/model/questions/question_slider.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/model/questions/question_rating.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/model/questions/question_text.dart';
+import 'package:acs_upb_mobile/widgets/radio_emoji.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+
+class FeedbackQuestionForm extends StatefulWidget {
+  const FeedbackQuestionForm({
+    this.question,
+    this.answerValues,
+    this.formKey,
+  });
+
+  final FeedbackQuestion question;
+  final List<Map<int, bool>> answerValues;
+  final GlobalKey<FormState> formKey;
+
+  @override
+  _FeedbackQuestionFormState createState() => _FeedbackQuestionFormState();
+}
+
+class _FeedbackQuestionFormState extends State<FeedbackQuestionForm> {
+  @override
+  Widget build(BuildContext context) {
+    if (widget.question is FeedbackQuestionSlider) {
```

Comment:
Actually, since you have separate classes for this, why not define a `Widget widget` getter in `FeedbackQuestion` and then implement it accordingly in each child class? It would make the code much easier to follow.

Alternatively, at least make separate methods for each type as this is way too long.